### PR TITLE
week-6-final-VMARIN

### PR DESCRIPTION
**Pull request by VMMarin**

### CODE REVIEW

**Review by elysemontano — CHANGES_REQUESTED** [review]
I need to see pseudocode for problem one in javascript challenges to continue this review.

**Comment by VMMarin** [comment]
Requested changes have been submitted.

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,38 +14,106 @@
...
+// Split array and capitalize (toUpperCase) 
+// return split and capitalized array as a sentence using string interpolation
+
+const capitalizeArr = (arr) => {
+  return arr.map(value => {
+    let splitArray = value.name.split(' ')
+    let capitalArr = splitArray[0][0].toUpperCase() +splitArray[0].substring(1) + " " + splitArray[1][0].toUpperCase() + splitArray[1].substring(1)
+    return `${capitalArr} is a ${value.occupation}`
+  })
+}
```
Comment (elysemontano) [inline code-challenges.test.js:47]: Logic and test looks good!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,38 +14,106 @@
...
+//use modulo  to find remainder
+//return new array
+
+const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+
+  const remainderFind = arr.filter(value(num){
+    return num % 3 === 0
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:87]: Does this pass your test?  Currently this does not look like it is a function so arr would be undefined.

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,38 +14,106 @@
...
+
+//convert array to string using .toString
+//math.hypot will return square root of sum
+//raise sum to power of 2
+
+const cubeAlicious = () => {
+  return arr.toString
+}
+  const thisVariable = Math.hypot(...cubeAlicious().split`,`)
+  return thisVariable**2
```
Comment (elysemontano) [inline code-challenges.test.js:119]: I have never heard of Math.hypot before, pretty cool!  Try to keep all of your code inside your function.  A function is there to be a reusable block of code that will provide a specific desired output.  Providing outputs outside the function may not have the same desired result.

File: interview-questions.md [hunk 4]
```diff
@@ -7,44 +7,112 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Your answer:
+  Your answer: Been there!  First, let's address what a foreign key is. A foreign key is a column in a relational database table that provides a link between data in two tables. It acts as a cross-reference between tables because it references the primary key of another table, thereby establishing a link between them. In this case, table Cohort has_many, students belong to Cohort. Every table has primary key - id:number.
+   The foreign key is reference to another table's primary key
 
-  Researched answer:
+  Researched answer: Been there!  First, let's address what a foreign key is. A foreign key is a column in a relational database table that provides a link between data in two tables. It acts as a cross-reference between tables because it references the primary key of another table, thereby establishing a link between them. In this case, table Cohort has_many, students belong to Cohort.Every table has primary key - id:number. The foreign key is reference to another table's primary key
+  Example of how to fix the issue
```
Comment (elysemontano) [inline interview-questions.md:14]: A model is created initially, but if you forgot to add the cohort_id when the model was initially created you would need to create a migration.

File: interview-questions.md [hunk 5]
```diff
@@ -7,44 +7,112 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+action: "PATCH"  location: /students/2   - update - modifying student id: 2
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2      destroy - remove student id: 2
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+- As a developer, I can create a class called Task.
```
Comment (elysemontano) [inline interview-questions.md:100]: These are all great developer stories.  User stories are usually geared more toward what the user experience will be vs what the developer will be creating.

File: rails-commenting.rb [hunk 6]
```diff
@@ -54,15 +54,15 @@ def destroy
...
       redirect_to blog_post_path(@post)
     end
   end
 
-  # ---9)
+  # ---9)anything under private will not be visible to user
   private
   def blog_post_params
-    # ---10)
+    # ---10) we're saying here that the paramters required to create a blog post are :title, :content
```
Comment (elysemontano) [inline rails-commenting.rb:65]: Great job on this section!

**Review by elysemontano — APPROVED** [review]
Last assessment is DONE (once merged of course 😀)!  Your interview answers were amazing and you crushed the rails section!  I would really recommend spending some time reviewing [javascript functions](https://github.com/learn-academy-2022-bravo/Syllabus/blob/main/javascript/functions.md) in the syllabus.  Knowing the syntax and why we utilize functions is going to be very important when you start interviewing.

**Pull request merged**